### PR TITLE
Added condition to prevent user from inserting the same product twice

### PR DESCRIPTION
**Pull request by BenHurCreations**
- in file script which is triggered on click by user 
- soon to add user interface to change quantity of certain product

### CODE REVIEW

File: .github/workflows/pylint.yml [hunk 1]
```diff
@@ -21,19 +21,19 @@ jobs:
...
       run: |
         pip3 install virtualenv
-        virtualenv venv --python=python3.8
-        source venv/bin/activate
-        pip3 install flask Flask-SQLAlchemy Flask-Bootstrap SQLAlchemy flask_login flask_wtf pymysql pytest pytest-ordering requests pylint
-        export FLASK_APP=superx/app.py
+        virtualenv venv --python=python3.8  
+        source venv/bin/activate  
+        pip3 install flask Flask-SQLAlchemy Flask-Bootstrap SQLAlchemy flask_login flask_wtf pymysql pytest pytest-ordering requests pylint 
+        export FLASK_APP=superx/app.py
```
Comment (simzacks) [inline .github/workflows/pylint.yml]: Can you explain why you added a space after each of the last 4 lines?

File: bootstrap.sh [hunk 2]
```diff
@@ -18,4 +18,9 @@ pip3 install flask Flask-SQLAlchemy Flask-Bootstrap SQLAlchemy flask_login flask
 
 export FLASK_APP=/vagrant/superx/app.py
 export FLASK_ENV=development
-flask run -h 0.0.0.0 -p 5000
+
+# No need to run the app on CI
+if [ ! "$CI" ]
```
Comment (simzacks) [inline bootstrap.sh]: If the app isn't running then how are the tests going to check the functionality?

File: superx/static/js/script.js [hunk 3]
```diff
@@ -1,25 +1,45 @@
 // JavaScripts functions
 
+// Check if item is already in cart
+function addItemIfInCart(product_id, product_name){
+  //test 1- console.log("yes");
+  let b = false;
```
Comment (simzacks) [inline superx/static/js/script.js]: do you know the difference between let and var? and are you purposefully using let?

File: .github/workflows/pylint.yml (hunk 1 above)
Comment (BenHurCreations) [reply .github/workflows/pylint.yml]: by mistake, corrected

File: superx/static/js/script.js (hunk 3 above)
Comment (BenHurCreations) [reply superx/static/js/script.js]: used the function in a different location which required let

recommited with to var

File: bootstrap.sh (hunk 2 above)
Comment (BenHurCreations) [reply bootstrap.sh]: Most of the checks check the DB and the backend functionality, which do not require the app running in the background

do you think we should run the app in CI as well?

Comment (simzacks) [reply bootstrap.sh]: I think your CI should be testing application functionality.
Did you verify that the tests actually work when the application is off?
What are you going to do if you add a new test for application functionality that requires the application to be on?

Comment (BenHurCreations) [reply bootstrap.sh]: all the current tests are working right now.
however I added it to the CI for future relevant tests
ready to be merged

**Comment by simzacks** [comment]
Looks like the CI died in the middle.

**Comment by BenHurCreations** [comment]
> Looks like the CI died in the middle.

@simzacks fixed for good, removed the app part for now, does not affect the tests

**Review by simzacks — APPROVED** [review]

**Pull request merged**